core/db/migrate/20201201170826_remove_archived_user_addresses.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class RemoveArchivedUserAddresses < ActiveRecord::Migration[5.2]
+  def up
+    Spree::UserAddress.where(archived: true).delete_all
```

Comment:
I'm still a little bit scared about this. We are assuming people are only using this field in the way we think but it's not guaranteed. 

Maybe we can keep the `archived` column (ignoring it on the Rails side) and use the `bin/rails g solidus:update` task to be sure people are understanding that those records will be deleted from their database. What do you think?

Reply:
@waiting-for-dev correct me if I'm wrong here, but if I recall correctly it's a matter of adding a new method in the [update generator](https://github.com/solidusio/solidus/blob/master/core/lib/generators/solidus/update/update_generator.rb), isn't it?

Reply:
I really can't imagine that - and if we leave the field, we'll have to deal with it in every upgrade from now on (similar to the `Spree::Shipment#deprecated_address_id` field. I'd rather add a large changelog entry, and be done with it. People who use the `archived` flag for something will still be able to edit the migration on their project to not delete the records / column.

Reply:
Using [ignored_columns](https://apidock.com/rails/v6.1.3.1/ActiveRecord/ModelSchema/ClassMethods/ignored_columns) I think it will be pretty transparent for us. 

In my opinion, it's not a matter of imagining how the feature could be used, but more of a formal process of making this software as compliant with SemVer as possible. And removing a table column in a minor release is definitely a breaking change from all possible points of view.

For `Spree::Shipment#deprecated_address_id` (https://github.com/solidusio/solidus/pull/4333), it is a bit different because we are not using that field in the code from 2016 as you mentioned in the PR, so it's very unlikely that it is being used. BTW, I think that PR should go in a major release as well. I didn't comment because I just missed it. 🙂

Reply:
> @waiting-for-dev correct me if I'm wrong here, but if I recall correctly it's a matter of adding a new method in the [update generator](https://github.com/solidusio/solidus/blob/master/core/lib/generators/solidus/update/update_generator.rb), isn't it?

Do you mean adding a message? Yes, we could do that by adding a new method. However, the actual code is already [redirecting people to the Changelog](https://github.com/solidusio/solidus/blob/547d4eab818796ae77d0bd743359865ab189d069/core/lib/generators/solidus/update/update_generator.rb#L66). I think it'd be better to add any information there and keep using it as the unique source of truth for upgrades info.

Reply:
Basically, this way we won't have to remember removing the new message before the next Solidus version :wink: